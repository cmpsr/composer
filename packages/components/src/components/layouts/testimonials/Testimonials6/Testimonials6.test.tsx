import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Testimonials6, TESTIMONIAL6_PLAN_DEFAULT_TEST_ID } from '.';

describe('Testimonials6', () => {
  const testimonial =
    'I realized up there that our planet is not infinite. Itâ€™s fragile.';
  const author = 'Testing author,';
  const profession = 'Testing profession.';

  it('should match snapshot', () => {
    const testimonial6 = renderer.create(
      <Testimonials6
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );

    const tree = testimonial6.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render component', () => {
    const { getByTestId } = render(
      <Testimonials6
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial6 = getByTestId(TESTIMONIAL6_PLAN_DEFAULT_TEST_ID);
    expect(testimonial6).toBeInTheDocument();
  });
  it('should render testimonial', () => {
    const { getByText } = render(
      <Testimonials6
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial6 = getByText(testimonial);
    expect(testimonial6).toBeInTheDocument();
  });
  it('should render author', () => {
    const { getByText } = render(
      <Testimonials6
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial6 = getByText(author);
    expect(testimonial6).toBeInTheDocument();
  });
  it('should render profession', () => {
    const { getByText } = render(
      <Testimonials6
        testimonial={testimonial}
        author={author}
        profession={profession}
      />,
    );
    const testimonial6 = getByText(profession);
    expect(testimonial6).toBeInTheDocument();
  });
});
