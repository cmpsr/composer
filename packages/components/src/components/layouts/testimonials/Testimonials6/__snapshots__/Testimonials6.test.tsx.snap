// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials6 should match snapshot 1`] = `
<div
  className="w-full h-full max-w-content mx-auto flex justify-center"
  data-testid="box"
>
  <div
    className="flex flex-col md:flex-row md:w-3/4 p-3"
    data-testid="testimonial6"
  >
    <svg
      className="md:w-20 h-12"
      data-testid="icon"
      height="50px"
      viewBox="0 0 24 24"
      width="50px"
    >
      <path
        d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"
      />
    </svg>
    <div>
      <h1
        className="typo-headline-2 p-1 text-center mb-10"
        data-testid="typography"
      >
        I realized up there that our planet is not infinite. Itâ€™s fragile.
      </h1>
      <div
        className="flex items-end justify-center"
      >
        <h6
          className="typo-headline-6 mr-2"
          data-testid="typography"
        >
          Testing author,
        </h6>
        <p
          className="typo-body text-gray-500"
          data-testid="typography"
        >
          Testing profession.
        </p>
      </div>
    </div>
  </div>
</div>
`;
